.github/workflows/publish-docker.yaml

```diff
@@ -0,0 +1,51 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: publish-docker
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  HUB: ghcr.io/apache/skywalking-python
+  PROJECT: skywalking-python
+
+jobs:
+  build-docker:
+    if: github.repository == 'apache/skywalking-python'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    timeout-minutes: 30
+    env:
+      VERSION: ${{ github.sha }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1.10.0
+        with:
+          registry: ${{ env.HUB }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push docker image
+        run: |
+          make push-image
```

Comment:
I tried to build and push locally and hours passed it is still running, I think we can parallize this by adding `make -j 10 push-image`?

Reply:
I have tried `make -j 10 push-image`, but seems that `docker buildx` can't run parallel.
Firstly, container `skywalking_python_main` can't be shared between parallel building tasks, secondly, even though we create target-specific containers, because [this line](https://github.com/jiang1997/skywalking-python/blob/a2584055130f5b8eb2f0c474bbac8213eef2094c/docker/Makefile#L81) works at system level, CI still runs into errors.

After some digging, I found that the most time-consuming step is building gRPC related whl packages when building image for arm64 `python3.11`. Cuz there is no binary whl package off the shelf so far. https://github.com/grpc/grpc/issues/32454

Reply:
I discovered that with the [`--builder`](https://docs.docker.com/engine/reference/commandline/buildx/#builder) option in Docker, we can make multiple targets simultaneously, but not to much(with `make -j 10` may run out of disk space).
Compared with no-parallel building, we can save about 10 minutes.